### ISSUE TITLE
can not open with refrainv

### ISSUE THREAD

**Issue by BriceKouadio** [viictorjs/Refrapy#3]
[for_tomopy.zip](https://github.com/viictorjs/Refrapy/files/7703986/for_tomopy.zip)
I Joined a travel time file from refrapick.

**Comment by viictorjs** [comment]
Hello @BriceKouadio 

I checked your file. It appears to contain first-break picks of only one seismic section, is it correct? I don't see the source position. Pick files from Refrapick have the following format:

number_of_shot_positions number_of_channels
position_of_first_receiver receiver_spacing
receiver1_position traveltime1 1
receiver2_position traveltime2 1
receiver3_position traveltime3 1
...
receiverN_position traveltimeN 1
/ source_position
...

Your file appears to be missing the source position row and the number of shot positions is 5, but 1 should be used in this case.

Assuming the source position as -20 m, your file should look like this:

1 24
0.00 2.00
0.000000 34.962191 1
2.000000 36.410901 1
4.000000 37.859611 1
6.000000 38.825417 1
8.000000 40.274127 1
10.000000 41.239933 1
12.000000 42.205739 1
14.000000 43.171546 1
16.000000 44.137352 1
18.000000 45.586062 1
20.000000 47.034771 1
22.000000 48.000578 1
24.000000 49.449287 1
26.000000 50.897997 1
28.000000 52.829610 1
30.000000 53.795416 1
32.000000 54.761222 1
34.000000 56.209932 1
36.000000 56.692835 1
38.000000 57.658642 1
40.000000 58.624448 1
42.000000 60.073158 1
44.000000 61.038964 1
46.000000 61.521867 1
/ -20.0

And this is the output *.sgt file created in Refrainv to try a traveltimes tomography :

25 # shot/geophone points
#x y
-20.00 0
0.00 0
2.00 0
4.00 0
6.00 0
8.00 0
10.00 0
12.00 0
14.00 0
16.00 0
18.00 0
20.00 0
22.00 0
24.00 0
26.00 0
28.00 0
30.00 0
32.00 0
34.00 0
36.00 0
38.00 0
40.00 0
42.00 0
44.00 0
46.00 0
24 # measurements
#s g t
1 2 0.034962
1 3 0.036411
1 4 0.037860
1 5 0.038825
1 6 0.040274
1 7 0.041240
1 8 0.042206
1 9 0.043172
1 10 0.044137
1 11 0.045586
1 12 0.047035
1 13 0.048001
1 14 0.049449
1 15 0.050898
1 16 0.052830
1 17 0.053795
1 18 0.054761
1 19 0.056210
1 20 0.056693
1 21 0.057659
1 22 0.058624
1 23 0.060073
1 24 0.061039
1 25 0.061522

**Comment by BriceKouadio** [comment]
![image](https://user-images.githubusercontent.com/70104501/145972214-caa6fe2e-9e1b-49fd-a757-5ab727c4812c.png)
Hi @viictorjs 

Thank you,

Yes, I didn't check the file before. But I don't know why it contains only one shot first-break picks as I picked all the shots before saving and you can see the image I joined to this comment. 
I tried different ways of saving the picks by trying to save picks shot by shot but whatever I do, it saves the same shot.

The missing shot position can be due to the fact that it's not the raw *.seg2 files I upload in refrapick.

maybe what I can do to overcome this issue is to upload shot by shot in refrapick, save each pick and gather all picks in on file by adding the shot coordinate. I will let you know

**Comment by BriceKouadio** [comment]
[pickallrp.zip](https://github.com/viictorjs/Refrapy/files/7710870/pickallrp.zip)

This file also can not be opened with refrainv. 
if the file is correct, the issue is on refrainv side. so maybe we should check dependencies. I use a new version of conda so I had downgrade some modules like matplotlib...

**Comment by viictorjs** [comment]
Hi @BriceKouadio 

I checked this new file. It appears to have all picked traveltimes from five sections, but there are some inconsistencies. Was this file created by Refrapick or did you manually edit it? If it was created by the picking program there must be some error when saving the output file, because it is not behaving as it should for the data set you are using.

This would be the correct version of your file:

5 24
0.00 2.00
0.000000 35.927998 1
2.000000 36.893804 1
4.000000 37.376707 1
6.000000 37.859611 1
8.000000 39.791223 1
10.000000 41.239933 1
12.000000 42.205739 1
14.000000 44.137352 1
16.000000 44.620255 1
18.000000 45.586062 1
20.000000 46.551868 1
22.000000 48.000578 1
24.000000 49.449287 1
26.000000 50.897997 1
28.000000 52.346706 1
30.000000 53.312513 1
32.000000 55.244126 1
34.000000 56.692835 1
36.000000 58.141545 1
38.000000 58.624448 1
40.000000 60.073158 1
42.000000 60.556061 1
44.000000 61.521867 1
46.000000 62.004770 1
/ -20
0.000000 0.696246 1
2.000000 7.342230 1
4.000000 13.197026 1
6.000000 19.368297 1
8.000000 25.381330 1
10.000000 31.077888 1
12.000000 32.185552 1
14.000000 33.609692 1
16.000000 35.033831 1
18.000000 36.141495 1
20.000000 37.723872 1
22.000000 38.831536 1
24.000000 39.939200 1
26.000000 41.679815 1
28.000000 43.103955 1
30.000000 44.528094 1
32.000000 45.793996 1
34.000000 47.059898 1
36.000000 47.692849 1
38.000000 48.484037 1
40.000000 49.591701 1
42.000000 50.699365 1
44.000000 51.490554 1
46.000000 51.965267 1
/ -1
0.000000 38.673299 1
2.000000 38.356823 1
4.000000 37.565635 1
6.000000 36.299733 1
8.000000 34.717356 1
10.000000 32.818503 1
12.000000 30.603175 1
14.000000 25.856043 1
16.000000 20.159485 1
18.000000 13.671739 1
20.000000 7.658706 1
22.000000 1.012721 1
24.000000 1.012721 1
26.000000 7.025755 1
28.000000 13.197026 1
30.000000 19.368297 1
32.000000 25.223092 1
34.000000 31.394364 1
36.000000 34.400880 1
38.000000 35.192069 1
40.000000 36.141495 1
42.000000 37.249159 1
44.000000 38.673299 1
46.000000 39.464487 1
/ 23
0.000000 53.009636 1
2.000000 52.267897 1
4.000000 51.278911 1
6.000000 50.784418 1
8.000000 50.042679 1
10.000000 48.064707 1
12.000000 46.086736 1
14.000000 44.850504 1
16.000000 44.108764 1
18.000000 43.367025 1
20.000000 42.625286 1
22.000000 41.389053 1
24.000000 39.905575 1
26.000000 38.916589 1
28.000000 38.174850 1
30.000000 36.196878 1
32.000000 35.207893 1
34.000000 34.466153 1
36.000000 31.499196 1
38.000000 25.318035 1
40.000000 19.878613 1
42.000000 13.450206 1
44.000000 7.269045 1
46.000000 4.549334 1
/ 47
0.000000 61.626174 1
2.000000 60.389942 1
4.000000 59.462768 1
6.000000 58.226536 1
8.000000 57.608420 1
10.000000 56.063129 1
12.000000 54.208781 1
14.000000 52.972549 1
16.000000 52.045375 1
18.000000 51.427259 1
20.000000 50.191027 1
22.000000 48.645736 1
24.000000 48.027620 1
26.000000 47.409504 1
28.000000 47.100446 1
30.000000 45.864214 1
32.000000 44.937040 1
34.000000 44.009866 1
36.000000 42.155517 1
38.000000 41.228343 1
40.000000 39.064937 1
42.000000 37.519647 1
44.000000 37.210589 1
46.000000 35.665298 1
/ 66

Please note that there must be an empty space between the "/" and the source position value. Also, your file was presenting an unnecessary row "0.00 2.00" when starting a new traveltimes block. The end of a block is represented by the source position, and a new receiver positions and traveltimes block must come right after.

**Comment by BriceKouadio** [comment]
Hi Victor, 

Thank you!
I have been able to run tomography and time term inversion! 
Now, I will look carefully at the parameters. I got less than 1ms as absolute error, with the velocity model I think it's good.

I will let you know if I find issues. 

Thank you!
![tomogrphy](https://user-images.githubusercontent.com/70104501/146395973-01a42312-0a31-4289-9d44-a348b64eb7a0.png)
![time_term_inver](https://user-images.githubusercontent.com/70104501/146395979-1ffe778a-42bc-4014-8571-df75af62b082.png)
